src/typescript/frontend/src/components/StatsBar.tsx

```diff
@@ -1,51 +1,324 @@
-      <div className="my-auto">
-        <WalletSelector />
+      <div className="my-auto hidden md:block">
+        <SocialMediaIcons />
       </div>
     </div>
   );
 }
+
+// MOCK FUNCTIONS + DATA
+export const useMarketData = (market: string): UseQueryResult<MarketStats> => {
+  return useQuery(
+    ["marketStats", market],
+    async () => {
+      // MOCK API CALL
+      const baseAsset = market.split("-")[0];
+      const quoteAsset = market.split("-")[1];
+      function timeout(ms: number | undefined) {
+        return new Promise((resolve) => setTimeout(resolve, ms));
+      }
+      await timeout(1000);
+      const {
+        lastPrice,
+        change24h,
+        high24h,
+        low24h,
+        pairData,
+        lastPriceChange,
+        change24hPercent,
+      } = STATS_BAR_MOCK_DATA;
+      // END MOCK API CALL
+      return {
+        lastPrice: formatDecimal(lastPrice),
+        lastPriceChange: formatDecimalWithPlusMinus(lastPriceChange, 4),
+        change24h: formatDecimalWithPlusMinus(change24h, 4),
+        change24hPercent: formatDecimalWithPlusMinus(change24hPercent),
+        high24h: formatDecimal(high24h, 4),
+        low24h: formatDecimal(low24h, 4),
+        pairData: {
+          baseAsset: baseAsset,
+          quoteAsset: quoteAsset,
+          baseAssetIcon: pairData.baseAssetIcon,
+          quoteAssetIcon: pairData.quoteAssetIcon,
+          baseVolume: formatDecimal(pairData.baseVolume),
+          quoteVolume: formatDecimal(pairData.quoteVolume),
+        },
+      } as MarketStats;
+    },
+    { keepPreviousData: true, refetchOnWindowFocus: false }
+  );
+};
+
+const STATS_BAR_MOCK_DATA: MarketStats = {
+  lastPrice: "10.17",
+  lastPriceChange: "10.1738",
+  change24h: "-.9305",
+  change24hPercent: "-8.38",
+  high24h: "11.1681",
+  low24h: "9.85",
+  pairData: {
+    // asset names here don't matter for testing purposes
+    baseAsset: "doesn't",
+    quoteAsset: "matter",
+    baseAssetIcon: "/tokenImages/APT.png",
+    quoteAssetIcon: "/tokenImages/USD.png",
+    baseVolume: "6531688.77",
+    quoteVolume: "68026950.84",
+  },
+};
+
+// UTIL FUNCTIONS
+
+// format number to dollar
+const formatDecimal = (num: string, digits = 2): string => {
+  const roundedNum = Number(num).toFixed(digits);
+  const parts = roundedNum.split(".");
+  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+  return parts.join(".");
+};
+
+// formatDecimal but with + or - sign
+const formatDecimalWithPlusMinus = (num: string, digits = 2): string => {
+  const formattedNum = formatDecimal(num, digits);
+  return Number(num) >= 0 ? `+${formattedNum}` : formattedNum;
+};
+
+// color based on number, green if positive, red if negative
+const colorBasedOnNumber = (num: string | undefined): string => {
+  if (!num) return "text-green-500";
+
+  if (num[0] === "+") return "text-green-500";
+  if (num[0] === "-") return "text-red-500";
+
+  return Number(num) < 0 ? "text-red-500" : "text-green-500";
+};
```

Comment:
I think it'll be easier if you just work off of the raw value that the API gives you, and do something like

```tsx
<span
  className={`${change24h < 0 ? "text-red-500" : "text-green-500"}`}
>
  {change24h < 0 ? "-" : "+"}change24h
</span>
```

This way, you get the benefit of the Tailwind plugins for eslint and prettier checking your classnames, whereas if you define classnames outside the context of a `className` attribute, they don't get checked.